Sources/kai/Checker.swift

```diff
@@ -1344,6 +1349,65 @@ extension Checker {
                 checkStmt(stmt)
             }
 
+        case .stmtSwitch(let subject, let cases, _):
+            let subjectType: Type?
+            
+            if let subject = subject {
+                subjectType = checkExpr(subject)
+            } else {
```

Comment:
The main reason I did this was so I could make subjectType immutable. But sure, updated to var.